### ISSUE TITLE
Upgrade to ROCm 5.6 to support RDNA3 consumer GPUs

### ISSUE THREAD

**Issue by hchasens** [oobabooga/text-generation-webui#3529]
**Description**

ROCm 5.6 brings support to a number of RDNA3 consumer GPUs. As the RX 7000 series is popular right now I think this would be a much wanted feature.

**Comment by jllllll** [comment]
ROCm 5.6 is definitely the more useful version of ROCm to support. However, Pytorch 2.0.1 does not support it. Only the nightly builds do at the moment. We can't use the nightly builds due to stability concerns. New builds are uploaded all the time and previous ones are not kept. This means that a broken build could be uploaded at any time and would break both the webui and the installer until a fix is uploaded. I actually encountered this just yesterday when a broken build was uploaded and completely broke one of my GitHub actions workflows.

**Comment by hchasens** [comment]
This is understandable and I figured this was why it hasn't been done yet. I suppose it's not worth changing the install script to ask the user if they'd want to use an unstable build since it'll probably be stable in a month or less. 

Out of curiosity, if I wanted to do it myself, is the only place where the ROCm version checked during the install script and in the Vert Env file? For my own home lab uses it'd be nice to not have to wait a month if all it takes are some small edits.

**Comment by jllllll** [comment]
The ROCm version is not checked at all as there are too many ways for such a check to not work.

If you want to change versions, then you need to install the version of Pytorch that you want to use using `cmd_linux.sh` and set the appropriate environment variables to point to your ROCm installation (`ROCM_PATH` and `HIP_PATH`, not sure if both are needed). There is a commented-out line near the top of `webui.py` that can be used to set `ROCM_PATH`.

Then, you need to build and install these packages from source using the newer ROCm:
```sh
export ROCM_PATH=/opt/rocm
export HIP_PATH=$ROCM_PATH
export PATH=/opt/rocm/bin:$PATH
export ROCM_VERSION=5.6
BUILD_CUDA_EXT=1 python -m pip install git+https://github.com/PanQiWei/AutoGPTQ@v0.4.0
python -m pip install git+https://github.com/jllllll/exllama
python -m pip install git+https://github.com/jllllll/GPTQ-for-LLaMa-CUDA
```

The exllama one can be easily skipped by uninstalling the package that is installed by the webui:
```
python -m pip uninstall exllama
```
This will result in the webui defaulting to the original JIT compiled version of ExLlama that is cloned into `text-generation-webui/repositories` by the installer. So long as ROCm is set up correctly, it will compile itself using ROCm next time you use ExLlama.

**Comment by DistantThunder** [comment]
Just a confirmation that this workaround does work on proper ROCm 5.6.x install.

**Comment by hchasens** [comment]
I just saw PyTorch now ships with ROCm 5.6 as part of it's stable package :)

**Comment by jllllll** [comment]
Yep. Webui now installs ROCm 5.6 packages as part of it's AMD GPU requirements.txt files.
